src/Symfony/Component/HttpFoundation/RedirectResponse.php

```diff
@@ -88,8 +88,21 @@ public function setTargetUrl($url)
 
         $this->targetUrl = $url;
 
-        $this->setContent(
-            sprintf('<!DOCTYPE html>
+        $this->setContent(sprintf($this->getHtmlRedirect(), htmlspecialchars($url, ENT_QUOTES, 'UTF-8')));
+
+        $this->headers->set('Location', $url);
+
+        return $this;
+    }
+
+    /**
+     * Return redirect html.
+     *
+     * @return string
+     */
+    protected function getHtmlRedirect() : string
```

Comment:
`getHtmlRedirect(string $url)` id say, or rely on `$this->targetUrl` being set.

What's the reason for this separation? :thinking:

Reply:
It's nice to have it separated because of readability.

Reply:
for readability it could also be a `HTML_REDIRECT_FORMAT` const :) currently it creates an extension point (protected) of which im not sure it's intended, or needed.

it's an implementation detail, IMHO the gain is minimal :)